avocado/utils/cpu.py

```diff
@@ -281,3 +282,49 @@ def set_cpuidle_state(state_number="all", disable=True, setstate=None):
                 except IOError as err:
                     logging.warning("Failed to set idle state on cpu %s "
                                     "for state %s:\n%s", cpu, state_no, err)
+
+
+def set_cpufreq_governor(governor="random"):
+    """
+    To change the given cpu frequency governor
+
+    :param governor: frequency governor profile name whereas `random` is default
+                     option to choose random profile among available ones.
+    """
+    avl_gov_file = "/sys/devices/system/cpu/cpu0/cpufreq/scaling_available_governors"
+    cur_gov_file = "/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor"
+    if os.access(avl_gov_file, os.R_OK) and os.access(cur_gov_file, os.W_OK):
+        avl_govs = open(avl_gov_file, 'r').read().strip().split(' ')
+        cur_gov = open(cur_gov_file, 'r').read().strip()
+        if governor == "random":
+            avl_govs.remove(cur_gov)
+            governor = avl_govs[random.randint(0, len(avl_govs)-1)]
```

Comment:
wouldn't `random.choice(avl_govs)` fit better?

Reply:
Sure, it is, thanks :-)